examples/hosts/app-integration/schema-upgrade/src/start.ts

```diff
@@ -0,0 +1,123 @@
+    location.hash = id;
+
+    // Put the ID in the tab title
+    document.title = id;
+};
+
+const isInventoryListContainer1 = (model: IVersionedModel): model is InventoryListContainer1 => {
+    return model.version === "one";
+};
+
+const isInventoryListContainer2 = (model: IVersionedModel): model is InventoryListContainer2 => {
+    return model.version === "two";
+};
+
+const render = (model: IVersionedModel) => {
+    const appDiv = document.getElementById("app") as HTMLDivElement;
+    ReactDOM.unmountComponentAtNode(appDiv);
+    // This demo uses the same view for both versions 1 & 2 - if we wanted to use different views for different model
+    // versions, we could check its version here and select the appropriate view.
+    // TODO: Better view code loading.
+    if (isInventoryListContainer1(model) || isInventoryListContainer2(model)) {
+        ReactDOM.render(
+            React.createElement(InventoryListContainerView, { model }),
+            appDiv,
+        );
+    } else {
+        throw new Error(`Don't know how to render version ${model.version}`);
+    }
+
+    // The DebugView is just for demo purposes, to manually control code proposal and inspect the state.
+    const debugDiv = document.getElementById("debug") as HTMLDivElement;
+    ReactDOM.unmountComponentAtNode(debugDiv);
+    ReactDOM.render(
+        React.createElement(DebugView, {
+            model,
+        }),
+        debugDiv,
+    );
+};
+
+async function start(): Promise<void> {
+    const tinyliciousService = new TinyliciousService();
+
+    const modelLoader = new ModelLoader<IMigratableModel>({
+        urlResolver: tinyliciousService.urlResolver,
+        documentServiceFactory: tinyliciousService.documentServiceFactory,
+        codeLoader: demoCodeLoader,
+        modelCodeLoader: new DemoModelCodeLoader(),
+        generateCreateNewRequest: createTinyliciousCreateNewRequest,
+    });
+
+    let id: string;
+    let model: IMigratableModel;
+
+    if (location.hash.length === 0) {
+        // Choosing to create with the "old" version for demo purposes, so we can demo the upgrade flow.
+        // Normally we would create with the most-recent version.
+        const createResponse = await modelLoader.createDetached("one");
+        model = createResponse.model;
+
+        // Fetching and importing the data here is optional
+        // For demo purposes it's nice to have some prepopulated entries though.
+        const fetchedData = await externalDataSource.fetchData();
+        if (!model.supportsDataFormat(fetchedData)) {
+            throw new Error("Model doesn't support fetched data format");
+        }
+        await model.importData(fetchedData);
+
+        id = await createResponse.attach();
+    } else {
+        id = location.hash.substring(1);
+        model = await modelLoader.loadExisting(id);
+    }
+
+    // TODO: Could be reasonable to merge Migrator into the ModelLoader, for a MigratingModelLoader.
```

Comment:
My two cents: I think I prefer it being separate for the purpose of this demo. It's nice that the `ModelLoader` can abstract away parts of the demo that don't relate to migrating container schema, so `Migrator` can focus on that.